.editorconfig

```diff
@@ -0,0 +1,6 @@
+# top-most EditorConfig file
+root = true
+
+# Unix-style newlines with a newline ending for *.el files (prevents QUELPA errors)
+[*.el]
+end_of_line = lf
```

Comment:
what's this file for? do we need to have it in the repo?

Reply:
Sorry for not mention this, the intention of this file is for enforcing new line to be always LF. This helps when contributor is coming from other OS that has different new line (CRLF or CR).
Quelpa somehow is quite sensitive to line ending when installing package, if the line ending of a package is not LF, Quelpa may fail to install it.
So having this file will ensure the uniform format for all of the file.
https://editorconfig.org/

Reply:
> Quelpa somehow is quite sensitive to line ending when installing package, if the line ending of a package is not LF, Quelpa may fail to install it.

This `.editorconfig` file has nothing to do with whether or how Quelpa installs Elisp packages.

Reply:
This file ensure that all files in this repository will have line ending with LF. Which makes Quelpa be able to install it.
I'm on system that has different line ending than LF, having this will ensure that the line ending is not accidentally changed, does that make sense?
Please see https://github.com/kiennq/lsp-powershell/pull/8#issuecomment-485110018

Reply:
So you're talking about Quelpa installing itself?

Reply:
> So you're talking about Quelpa installing itself?
> 
> Anyway, that file does not ensure anything, because it doesn't take effect unless the user editing the file is using `editorconfig`.

Yes, you're correct. I'm talking about installing Quelpa with Quelpa.



> Looking at that issue, it has nothing to do with adding an editorconfig file here. Adding one here will not affect how Quelpa installs other packages that have CRLF endings.

Yes, that's separate issue. It won't solve the problem of Quelpa installs **other** packages. We can perhaps investigating on that one, but that's eventually different issue.

> If any files in Quelpa have CRLF endings, we should fix that. If your Emacs config produces CRLF endings, you should fix that in your config. But adding this file is not necessary and won't prevent CRLF endings.

I'm disagree with this. This file will help when other creates PR for this repo.
1. Github doesn't clearly suggest line ending is changed. You saw the change highlighted but it's difficult to see that actually the line ending is changed.
2. It's standard to use `.editorconfig` for this kind of stuff, especially when there're multiple contributors. Quick search on github yield more than [7M+ code results about that](https://github.com/search?q=in%3Apath+.editorconfig&type=Code), even github itself supports editorconfig.
3. It's actually difficult to see why thing goes wrong even with CI. You will see build failing, but without clear indicator why.